src/renderer/components/dock/upgrade-chart/create-upgrade-chart-tab.injectable.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable, lifecycleEnum } from "@ogre-tools/injectable";
+import upgradeChartTabStoreInjectable from "./store.injectable";
+import dockStoreInjectable from "../dock/store.injectable";
+import type { HelmRelease } from "../../../../common/k8s-api/endpoints/helm-releases.api";
+import { DockStore, DockTabCreateSpecific, TabId, TabKind } from "../dock/store";
+import type { UpgradeChartTabStore } from "./store";
+import { bind } from "../../../utils";
+import { runInAction } from "mobx";
+
+interface Dependencies {
+  upgradeChartStore: UpgradeChartTabStore;
+  dockStore: DockStore
+}
+
+function createUpgradeChartTab({ upgradeChartStore, dockStore }: Dependencies, release: HelmRelease, tabParams: DockTabCreateSpecific = {}): TabId {
```

Comment:
Notice: Name of the function lies. If tab already exists, it opens instead of creating. If it doesn't it creates it (and I imagine that something else opens it?)